### PR TITLE
Add doc section for auth media

### PR DESCRIPTION
**Pull request by devonh**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -181,3 +181,18 @@ Example (generated using the body and public key from the previous examples):
     }
 }
 ```
+
+## Authenticated Media
+
+When accessing media from a Synapse homeserver with authenticated media enabled, an additional
+HTTP header needs to be passed along with any request to the Matrix Content Scanner.
+
+This header follows the Matrix specification `Authentication Bearer scheme` as outlined [here](https://spec.matrix.org/v1.12/client-server-api/#using-access-tokens).
```
Comment (anoadragon453) [inline docs/api.md:190]: The Authentication Bearer scheme is credited to web standards, rather than the Matrix spec. We just use it.

```suggestion
This header follows the `Authentication Bearer scheme` as [outlined in the Matrix specification](https://spec.matrix.org/v1.12/client-server-api/#using-access-tokens).
```

File: docs/api.md [hunk 2]
```diff
@@ -181,3 +181,18 @@ Example (generated using the body and public key from the previous examples):
     }
 }
 ```
+
+## Authenticated Media
+
+When accessing media from a Synapse homeserver with authenticated media enabled, an additional
+HTTP header needs to be passed along with any request to the Matrix Content Scanner.
```
Comment (anoadragon453) [inline docs/api.md:188]:
```suggestion
When accessing media from a Synapse homeserver with authenticated media enabled, an
`Authorization` HTTP header must be passed along with any request to the Matrix Content Scanner.
```

File: docs/api.md [hunk 3]
```diff
@@ -181,3 +181,18 @@ Example (generated using the body and public key from the previous examples):
...
+
+This header follows the Matrix specification `Authentication Bearer scheme` as outlined [here](https://spec.matrix.org/v1.12/client-server-api/#using-access-tokens).
+The `Authorization` header method must be used, sending the access token as a query string
+parameter is not supported.
+
+Example authorization header:
+
+```
+Authorization: Bearer <access_token>
+```
```
Comment (anoadragon453) [inline docs/api.md:198]: It may be worth quickly saying what HTTP status code and body will be returned if the header is not present or the token is invalid?

File: docs/api.md [hunk 4]
```diff
@@ -181,3 +181,18 @@ Example (generated using the body and public key from the previous examples):
...
 }
 ```
+
+## Authenticated Media
+
+When accessing media from a Synapse homeserver with authenticated media enabled, an additional
+HTTP header needs to be passed along with any request to the Matrix Content Scanner.
+
+This header follows the Matrix specification `Authentication Bearer scheme` as outlined [here](https://spec.matrix.org/v1.12/client-server-api/#using-access-tokens).
+The `Authorization` header method must be used, sending the access token as a query string
```
Comment (anoadragon453) [inline docs/api.md:191]: Folks might be confused about *what* access token they should use. You may want to say that it should be the Matrix access token of the client's user.

File: pyproject.toml [hunk 5]
```diff
@@ -69,7 +69,7 @@ version = "1.1.0"
 description = "A web service for scanning media hosted by a Matrix media repository"
 authors = ["Element Backend Team <team-backend-synapse@element.io>"]
 readme = "README.md"
-license = "AGPL-3.0-only"
+license = "AGPL-3.0-only OR LicenseRef-Element-Commercial"
```
Comment (anoadragon453) [inline pyproject.toml:72]: Could you put this in a separate PR? It seems unrelated to the doc change.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
A few things here and there, but overall on track!

File: pyproject.toml (hunk 5 above)
Comment (devonh) [reply pyproject.toml]: Moved here: https://github.com/element-hq/matrix-content-scanner-python/pull/71

File: docs/api.md [hunk 6]
```diff
@@ -181,3 +181,26 @@ Example (generated using the body and public key from the previous examples):
...
+parameter is not supported.
+
+Example authorization header:
+
+```
+Authorization: Bearer <access_token>
+```
+
+If a request is made for authenticated media and the access token is invalid, the content scanner
+will responde with a `MCS_MEDIA_REQUEST_FAILED`.
```
Comment (anoadragon453) [inline docs/api.md:204]:
```suggestion
will respond with HTTP status 502, errcode `MCS_MEDIA_REQUEST_FAILED`.
```

ref: https://github.com/element-hq/matrix-content-scanner-python/blob/694914f92bf100a37f7e1baee527cfb3cd01303f/src/matrix_content_scanner/scanner/file_downloader.py#L210-L231

Seems like we should probably be returning a 401 here though, in line with other Matrix-esque endpoints? (This doesn't need to be fixed here, but if you agree then we can track this in a separate issue.)

File: docs/api.md [hunk 7]
```diff
@@ -181,3 +181,26 @@ Example (generated using the body and public key from the previous examples):
...
+If a request is made for authenticated media and the `Authorization` header is missing, the content
+scanner will responde with a `M_NOT_FOUND`.
```
Comment (anoadragon453) [inline docs/api.md:206]:
```suggestion
scanner will respond with HTTP status 404, errcode `M_NOT_FOUND`.
```

**Review by anoadragon453 — APPROVED** [review]
LGTM! Just a couple tiny suggestions below.

File: docs/api.md (hunk 6 above)
Comment (devonh) [reply docs/api.md]: Yup yup - this should be changed to be more helpful for clients to understand what is going wrong.
Issue: https://github.com/element-hq/matrix-content-scanner-python/issues/72
